indico/modules/categories/controllers/display.py

```diff
@@ -670,3 +671,27 @@ def _get_event_data(self, event_query):
     def _render_ongoing_events(self, ongoing_events):
         template = get_template_module('categories/display/_calendar_ongoing_events.html')
         return template.render_ongoing_events(ongoing_events, self.category.display_tzinfo)
+
+
+class RHCategoryUpcomingEvent(RHDisplayCategoryBase):
+    """Redirect to the upcoming event of a category."""
+
+    def _process(self):
+        event_id = self._get_upcoming_event()
+        if event_id:
+            resp = redirect(url_for('events.display', confId=event_id))
```

Comment:
Return the event instead of its ID and you can use `event.url` here.

Also, we have some extra magic in our custom `url_for` wrapper that checks for a `locator` property on the object so you can also do something like `url_for('events.display', some_event)` because the locator returns `{'confId': self.id}` for an event.